zcash_client_sqlite/src/wallet/init.rs

```diff
@@ -106,24 +121,48 @@ pub fn init_wallet_db<P>(wdb: &WalletDb<P>) -> Result<(), rusqlite::Error> {
         )",
         NO_PARAMS,
     )?;
+    wdb.conn.execute(
+        "CREATE TABLE IF NOT EXISTS utxos (
+            id_utxo INTEGER PRIMARY KEY,
+            address TEXT NOT NULL, 
+            prevout_txid BLOB NOT NULL, 
+            prevout_idx INTEGER NOT NULL, 
```

Comment:
Do we not store a record in the `transaction` table for received transparent outputs? Because this table should be designed as the transparent analogue of `received_notes`, which uses:
```
            tx INTEGER NOT NULL,
            output_index INTEGER NOT NULL,
```
Doing this would also mean we don't need to store `height` per-UTXO, instead storing it per-tx.

---

That being said, given that auto-shielding has already been deployed, we can't easily change this now 😢

Reply:
It's not that hard to fix; we could just add the foreign key column with the reference to the transactions table in a migration that creates it as nullable, populates it by looking up by txid, and then changes the constraint to be not-null, and the `prevout_idx` column name is purely internal.

Reply:
Address after https://github.com/zcash/librustzcash/issues/369

Reply:
Opened https://github.com/zcash/librustzcash/issues/503 for this.